igor-web/src/main/groovy/com/netflix/spinnaker/igor/wercker/WerckerService.groovy

```diff
@@ -0,0 +1,349 @@
+                        kv("runId", run.id), kv("url", run.url),
+                        kv("buildNumber", runIdBuildNumbers.get(run.id)))
+
+                //return the integer build number for this run id
+                return runIdBuildNumbers.get(run.id)
+            } catch (RetrofitError e) {
+                def body = e.getResponse().getBody()
+                String wkrMsg
+                if (body instanceof TypedByteArray) {
+                    wkrMsg = new String(((retrofit.mime.TypedByteArray) body).getBytes())
+                } else {
+                    wkrMsg = body.in().text
+                }
+                log.error("Failed to trigger build for pipeline {}. {}", kv("pipelineName", pipelineName), kv("errMsg", wkrMsg))
+                throw new BuildController.BuildJobError(
+                "Failed to trigger build for pipeline ${pipelineName}! Error from Wercker is: ${wkrMsg}")
+            }
+        } else {
+            throw new BuildController.InvalidJobParameterException(
+            "Could not retrieve pipeline ${pipelineName} for application ${appName} from Wercker!")
+        }
+    }
+
+    /**
+     * Returns List of all Wercker jobs in the format of type/org/app/pipeline 
+     */
+    public List<String> getJobs() {
+        List<String> jobs = []
+        long start = System.currentTimeMillis();
+        List<Application> apps = getApplications();
+        start = System.currentTimeMillis()
+        apps.each { app ->
+            try {
+                List<Pipeline> pipelines = app.pipelines? app.pipelines : []; //getPipelines(app.owner.name, app.name);
+                jobs.addAll( pipelines.collect {
+                    it.type + SPLITOR + app.owner.name + SPLITOR + app.name + SPLITOR + it.name } )
+            } catch(retrofit.RetrofitError err) {
+                log.error "Error getting pipelines for ${app.owner.name } ${app.name} ${err}"
+            }
+        }
+        log.debug "getPipelines: ${jobs.size()} pipelines in ${System.currentTimeMillis() - start}ms"
+        return jobs
+    }
+
+    List<Application> getApplications() {
+        new SimpleHystrixCommand<List<Application>>(groupKey, buildCommandKey("getApplications"), {
+            return werckerClient.getApplications(authHeaderValue, limit)
+        }).execute();
+    }
+
+    List<Pipeline> getPipelines(String org, String app) {
+        new SimpleHystrixCommand<List<Pipeline>>(groupKey, buildCommandKey("getPipelines"), {
+            return werckerClient.getPipelinesForApplication(authHeaderValue, org, app)
+        }).execute();
+    }
+
+    /**
+     * A CommandKey should be unique per group (to ensure broken circuits do not span Wercker masters)
+     */
+    private String buildCommandKey(String id) {
+        return "${groupKey}-${id}"
+    }
+
+    List<Run> getBuilds(String appAndPipelineName) {
+        String[] split = appAndPipelineName.split(SPLITOR)
```

Comment:
I think most of this function can be replaced by a call to `getPipelineId`

Reply:
you are correct, fixed